packages/driver/src/cypress/command_queue.ts

```diff
@@ -60,336 +60,322 @@ const commandRunningFailed = (Cypress, state, err) => {
   })
 }
 
-export default {
-  create: (state, timeout, whenStable, cleanup, fail, isCy) => {
-    const queue = $queue.create()
-
-    const { get, slice, at, reset, clear, stop } = queue
-
-    const logs = (filter) => {
-      let logs = _.flatten(_.invokeMap(queue.get(), 'get', 'logs'))
+export class CommandQueue extends Queue<Command> {
+  state: any
+  timeout: any
+  whenStable: any
+  cleanup: any
+  fail: any
```

Comment:
Should this be of type func?

Reply:
Precisely, they are `$Cy['timeout']`, `$Cy['cleanup']`, `$Cy['fail']`. But the problem is that it creates a circular reference problem because `command_queue` is required in `cy.ts`. 

The goal of this PR is migrating functions inside `_.extend(cy, {})` to `$Cy` class. So, I left them out for the later PR.